L1Trigger/L1THGCal/interface/hw_interfaces/HGCalHardwareInterface.h

```diff
@@ -0,0 +1,59 @@
+#ifndef __L1Trigger_L1THGCal_HGCalHardwareInterface_h__
+#define __L1Trigger_L1THGCal_HGCalHardwareInterface_h__
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "DataFormats/L1THGCal/interface/HGCalCluster.h"
+#include "DataFormats/L1THGCal/interface/HGCalMulticluster.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+class HGCalHardwareInterface{
+  
+ public:
+  /* mapping is retrieved from: https://twiki.cern.ch/twiki/bin/viewauth/CMS/HGCALTriggerArchitectureDesignAndImplementation */
+  struct ClusterDataFormat{
+    enum CDF {ET = 0, CenterY = 8, CenterX = 20, Flags = 32, SizeY = 38, SizeX = 46, LocalMax = 54, NoTriggerCells = 56, LocalMax1RelX = 64,  LocalMax0ET = 72, LocalMax0RelY = 80, LocalMax0RelX = 88, LocalMax2RelY = 96,  LocalMax2RelX = 104,  LocalMax1ET = 112,  LocalMax1RelY = 120,  LocalMax3ET = 128, LocalMax3RelY = 136,  LocalMax3RelX = 144,  LocalMax2ET = 152 };
+  };
+  struct MulticlusterDataFormat{
```

Comment:
The main problem is that such a scheme prevents us from distinguishing the hardware levels: As you have stated previously 2D and 3D clusters are algorithm specific rather than hardware specific. I totally agree with declaring explicitly the 2D and 3D clusters, but we should have another abstraction level indicating from which level the information is arriving from. In the end, the two layer difference will be there independent of the 2D and 3D algorithms and containers.